### PR TITLE
✨ add in support for blockchain tests

### PR DESCRIPTION
**Pull request by zander-xyz**

### CODE REVIEW

File: test/ethereum_test/src/general_state_test.cpp [hunk 1]
```diff
@@ -14,6 +14,33 @@ using namespace monad;
 
 namespace
 {
+    [[nodiscard]] constexpr BlockHeader
+    block_header_from_env(nlohmann::json const &json)
```
Comment (rgarc) [inline test/ethereum_test/src/general_state_test.cpp:18]: Why was this moved from `from_json.hpp` ?

Comment (zander-xyz) [reply test/ethereum_test/src/general_state_test.cpp:18]: a block header in general state tests is different from a block header in blockchain tests. because of how it's labeled in general state tests, i separated the impl
